### ISSUE TITLE
[4.0.0-RC3]: Switch between entry types in entry editor maybe broken

### ISSUE THREAD

**Issue by kringkaste** [craftcms/cms#11093]
### Description

When you have at least two entry types with more than one tab and switch between them in the editor page, the new draft with the new layout is loaded but the tabs are not clickable / switchable anymore. 

### Steps to reproduce

1. Create a section with two entry types.
2. Add to both entry types at least two tabs with some fields.
3. Go to Entries and add a new entry with the default entry type.
4. Safe the entry.
5. Edit the same entry and switch the entry type in the details.

### Expected behavior

The field layout is loaded and switching between the tabs is possible.

### Actual behavior

The field layout is loaded but you cannot switch between the tabs anymore. The content of the active tab is always shown.

### Craft CMS version

4.0.0-RC3

### PHP version

8.0

### Installed plugins and versions

-

**Comment by kringkaste** [comment]
Reproducible on 4.0.0.1

**Comment by brandonkelly** [comment]
Thanks for reporting! Fixed for the next release.

**Comment by brandonkelly** [comment]
Craft 4.0.1 is out now with the fix for this.

**Comment by kringkaste** [comment]
I'm sorry, but the problem still exists :( I updated to 4.0.1 and  deleted cpresources. There are now new errors in the console: 
![2022-05-06T15:24:25,931364771+02:00](https://user-images.githubusercontent.com/964698/167140379-06b9c448-d431-4834-b558-64dd3b610f3a.png)

**Comment by brandonkelly** [comment]
I’m not able to reproduce that error, but see one place where it could have maybe occurred in `Tabs.js`, so just fixed that.

Can you try updating and let me know if it helps? Open `composer.json` and change your `craftcms/cms` requirement to `"dev-release/4.0.2 as 4.0.2"`, then run `composer update`.

**Comment by kringkaste** [comment]
I did some more testing and think I figured out the problem: As long as there is a tab with the same name in each of the entry types being switched between, there is no error. But if there is no tab with the same name, then no tab is selected when switching and then the check in line 114 in Tabs.js and the deselect in Cp.js in line 580 fails. So, test with two entry types with different tab names may reproduce this error.
